src/visualization/types.jl

```diff
@@ -348,58 +348,77 @@ end
+      for j in eachnode(dg), i in eachnode(dg)
+        u_node = get_node_vars(u, equations, dg, i, j, element)
+        u_extracted[sk, element] = u_node
+        sk += 1
+      end
+    end
+    # interpolate to volume plotting points
+    xplot, yplot = plotting_interp_matrix*x, plotting_interp_matrix*y
+    uplot = StructArray{SVector{nvars, uEltype}}(map(x->plotting_interp_matrix*x,
+                                                   StructArrays.components(u_extracted)))
 
-  # construct a triangulation of the plotting nodes
-  t = reference_plotting_triangulation((r_plot, s_plot))
+    xfp, yfp, ufp = mesh_plotting_wireframe(u_extracted, mesh, equations, dg, cache; nvisnodes=nvisnodes)
 
-  # extract x,y coordinates and solutions on each element
-  uEltype = eltype(u)
-  nvars = nvariables(equations)
-  x = reshape(view(cache.elements.node_coordinates, 1, :, :, :), n_nodes_2d, n_elements)
-  y = reshape(view(cache.elements.node_coordinates, 2, :, :, :), n_nodes_2d, n_elements)
-  u_extracted = StructArray{SVector{nvars, uEltype}}(ntuple(_->similar(x, (n_nodes_2d, n_elements)), nvars))
-  for element in eachelement(dg, cache)
-    sk = 1
-    for j in eachnode(dg), i in eachnode(dg)
-      u_node = get_node_vars(u, equations, dg, i, j, element)
-      u_extracted[sk, element] = u_node
-      sk += 1
-    end
-  end
+    transform_to_solution_variables!(uplot, solution_variables_, equations)
+    transform_to_solution_variables!(ufp, solution_variables_, equations)
 
-  # interpolate to volume plotting points
-  xplot, yplot = plotting_interp_matrix*x, plotting_interp_matrix*y
-  uplot = StructArray{SVector{nvars, uEltype}}(map(x->plotting_interp_matrix*x,
-                                                   StructArrays.components(u_extracted)))
+    return PlotData2DTriangulated(xplot, yplot, uplot, t, xfp, yfp, ufp, variable_names)
 
-  xfp, yfp, ufp = mesh_plotting_wireframe(u_extracted, mesh, equations, dg, cache; nvisnodes=nvisnodes)
+  # In the 3D-case a 'PlotData2DCartesian' object is created.
+  elseif ndims(mesh) == 3
 
-  # convert variables based on solution_variables mapping
-  solution_variables_ = digest_solution_variables(equations, solution_variables)
-  variable_names = SVector(varnames(solution_variables_, equations))
+    # Set nvisnodes to 32 on default.
+    if nvisnodes == nothing
+      nvisnodes = 32
+    end
 
-  transform_to_solution_variables!(uplot, solution_variables_, equations)
-  transform_to_solution_variables!(ufp, solution_variables_, equations)
+    # Convert data.
+    nodes = cache.elements.node_coordinates
+    data_on_plane, plane = unstructured_3d_to_2d_plane(nodes, u; elevations, slice, point, nvisnodes)
 
-  return PlotData2DTriangulated(xplot, yplot, uplot, t, xfp, yfp, ufp, variable_names)
+    return PlotData2DCartesian(vec(plane[1, 1, :]), vec(plane[2, :, 1]), [data_on_plane], variable_names, nothing, nothing, 0, 0)
+  else
+    @assert false "Input must be two- or three-dimensional."
```

Comment:
```suggestion
    throw(ArgumentError("Input must be two- or three-dimensional."))
```
`@assert` may be disabled
```julia
help?> @assert
  @assert cond [text]

  Throw an AssertionError if cond is false. Preferred syntax for writing assertions. Message text is optionally
  displayed upon assertion failure.

  │ Warning
  │
  │  An assert might be disabled at various optimization levels. Assert should therefore only be used as
  │  a debugging tool and not used for authentication verification (e.g., verifying passwords), nor
  │  should side effects needed for the function to work correctly be used inside of asserts.
```